src/components/PaymentStripe/checkoutForm.js

```diff
@@ -0,0 +1,370 @@
+    const cardElementOptions = {
+        style: {
+            base: {
+                iconColor: theme.palette.text.main,
+                fontFamily: '"Lato", Arial, sans-serif',
+                backgroundColor: theme.palette.background.main,
+                color: theme.palette.text.main,
+                fontSize: '1.1rem',
+                '::placeholder': {
+                    color: theme.palette.text.header,
+                    fontVariant: 300,
+                },
+            },
+        },
+    };
+    const handleInputChange = newDetail => {
+        const newbillingDetails = {
+            ...billingDetails,
+            ...newDetail,
+        };
+        setBillingDetails(newbillingDetails);
+        onChange({ billingDetails: newbillingDetails, cardDetails });
+    };
+    const handleAddressChange = newDetail => {
+        const cardElement = inline ? elements.getElement(CardElement) : null;
+        const addressComponents = newDetail.address ? newDetail.address.address_components : null;
+        const lastAddressComponent = addressComponents
+            ? addressComponents[addressComponents.length - 1]
+            : null;
+        const type = lastAddressComponent ? lastAddressComponent.types[0] : null;
+        if (type === 'postal_code') {
+            const newbillingDetails = {
+                ...billingDetails,
+                ...newDetail,
+                zipCode: lastAddressComponent.long_name,
+            };
+            if (cardElement) {
+                cardElement.update({
+                    value: { postalCode: lastAddressComponent.long_name },
+                });
+            }
+            handleInputChange({ ...newbillingDetails });
+            return;
+        }
+        if (cardElement) {
+            cardElement.update({
+                value: { postalCode: '' },
+            });
+        }
+        handleInputChange({
+            ...newDetail,
+            zipCode: '',
+        });
+    };
+    const handleCardElementChange = event => {
+        setError('');
+        const newbillingDetails = {
+            ...billingDetails,
+            address: {
+                ...billingDetails.address,
+                postal_code: event.value ? event.value.postalCode : '',
+            },
+            zipCode: inline && event.value ? event.value.postalCode : billingDetails.zipCode,
+        };
+        setBillingDetails(newbillingDetails);
+        setCardDetails({ ...event });
+        if (event.error) {
+            setError(event.error.message);
+        }
+        onChange({ billingDetails: newbillingDetails, cardDetails: event });
+    };
+    const handleSubmit = async event => {
+        event.preventDefault();
+        if (!stripe || !elements || !getPaymentIntent || !onProcessed) {
+            return;
+        }
+        try {
+            setLoading(true);
+            // TODO: validate fields and normalize address
+            const response = await getPaymentIntent();
```

Comment:
paymentIntent are made from server-side, what is the use case for this in the component?